services/skipchain/api.go

```diff
@@ -0,0 +1,130 @@
+package skipchain
+
+import (
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// Client is a structure to communicate with the Skipchain
+// service from the outside
+type Client struct {
+	*sda.Client
+}
+
+// NewClient instantiates a new client with name 'n'
+func NewClient() *Client {
+	return &Client{Client: sda.NewClient("Skipchain")}
+}
+
+// CreateRootInter creates two Skipchains: a root SkipChain with
+// maximumHeight of maxHRoot and an intermediate SkipChain with
+// maximumHeight of maxHInter. It connects both chains for later
+// reference.
+func (c *Client) CreateRootInter(elRoot, elInter *sda.EntityList, maxHRoot, maxHInter int, ver VerifierID) (root, inter *SkipBlock, err error) {
```

Comment:
I guess I'm not familiar with the code enough to deeply understand the design of it. However regarding the "api", I don't understand why we fix with this 3-level structure and don't have more generic structure that would allow greater flexibility. Again, I'm not so familiar so I might be wrong but I'd love to see the api more simpler, stg along the lines of:
```
// Create a skipchain with maximum height, and which
// is of type `chainType` which represents which type of 
// skipchain it is -> Roster Skipchain, or Timevault Skipchain,
// DNS Skipchain, or pure bytes Skipchain, etc.
func (c *Client) CreateSkipchain( height, chainType, ...  ) { ... }

// Appending a block to the skipchain,
// The block can contain a Roster or pure Bytes or whatever, 
// It all depends on the type of the Skipchain.
func (c *Client) AddBlock( chainID, data ) { ... }
 
// DERIVATION of AddData
func (c *Client) AddRoster( chainID, roster) { 
     ...
     c.AddBlock(chainID,rosterBlock)
 }
func (c *Client) AddData( chainID, data) { 
   ...
   c.AddBlock(chainID, dataBlock)
 }

// Add a parent-child link between two skipchains. 
// That creates the relationship between two skipchain 
// so one is the parent of the other.
func (c *Client) AddParentChildLink( parentID, childID ) { ... }
```
With a structure like this, it's easy to produce our 3-level layout but it's also easy to maintain one skiplist only easily (as we want to do for stuff like CONIKS and so on), or a 2-level layout as we might not always need an intermediate.

Again, it's very abstract design and I'm not so much familiar with the code yet, **and there may be subtleties I did not get**, but in general a simpler design leads to a better outcome ! 
Of course, it's not something that have to be done now, but depending on what you all think (feedbacks please!), it might be a good goal for a next iteration on this complex subject...